hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicOzoneClientAdapterImpl.java

```diff
@@ -210,10 +210,14 @@ protected void incrementCounter(Statistic objectsRead) {
   }
 
   @Override
-  public OzoneFSOutputStream createFile(String key, boolean overWrite,
+  public OzoneFSOutputStream createFile(String key, short replication, boolean overWrite,
       boolean recursive) throws IOException {
     incrementCounter(Statistic.OBJECTS_CREATED);
     try {
+      if (replication == ReplicationFactor.ONE.getValue()
+          || replication == ReplicationFactor.THREE.getValue()) {
+        replicationFactor = ReplicationFactor.valueOf(replication);
```

Comment:
`replicationFactor` can now be a local variable.

Reply:
The replicationFactor is initialized in the constructor. When user-set Replication is invalid (neither 1 nor 3), the previous default is used.

Reply:
Please correct me, but it looks like requests with valid `replication` will override the default value for later invalid requests.  Let's say that the cluster has `replicationFactor=3` configured and create requests with the following `replication` values are issued:

1. `replication=2`
2. `replication=1`
3. `replication=2`

The requests in step 1 and 3 will behave differently, since `replicationFactor` was changed by the request in step 2.

(It could be argued that requests with `replication=2` or other invalid value should be rejected, but that's another topic.)

Reply:
Yes, you're right. This is a place where it is possible to override the default replication. I'm going to add a local variable here to avoid this problem.